src/Perpetuum/Services/EventServices/EventProcessors/EnvironmentalEffectHandler.cs

```diff
@@ -59,47 +59,8 @@ private void OnStateChange()
                 return;
 
             ZoneEffect nextEffect = null;
-            if (_gameTime.IsDay)
-            {
-                if (_weatherState.IsGoodWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_day_clear);
-                }
-                else if (_weatherState.IsBadWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_day_overcast);
-                }
-                else
-                {
-                    nextEffect = GetEffect(EffectType.effect_day);
-                }
-            }
-            else if (_gameTime.IsNight)
-            {
-                if (_weatherState.IsGoodWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_night_clear);
-                }
-                else if (_weatherState.IsBadWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_night_overcast);
-                }
-                else
-                {
-                    nextEffect = GetEffect(EffectType.effect_night);
-                }
-            }
-            else
-            {
-                if (_weatherState.IsGoodWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_weather_good);
-                }
-                else if (_weatherState.IsBadWeather)
-                {
-                    nextEffect = GetEffect(EffectType.effect_weather_bad);
-                }
-            }
+
+            nextEffect = GetEffect(_gameTime.GetDayState(), _weatherState.getWeatherState());          
```

Comment:
Any reason to not simply declare and initialize `nextEffect` here? (why have ln61?)

Reply:
`var nextEffect = GetEffect(_gameTime.GetDayState(), _weatherState.getWeatherState());` basically.
^ var can be used because its no longer initialized with null.

Style nit: Also there is some trailing whitespace on this line. (I finally pulled the code!)